### DISCUSSION TITLE
Is it possible create `torch.Tensor` from `IDirect3DSurface` or CUDA memory directly?

### DISCUSSION THREAD

**Discussion by GF-Huang** [xamarin/TorchSharp#1238] [Q&A]
In details, I use Windows Graphics Capture technology to capture specific app window frames. This is equivalent to screen recording, but only for a specific window. You can read this [repo](https://github.com/robmikh/ManagedScreenshotDemo/blob/master/ManagedScreenshotDemo/Program.cs) if you interest it.

At currently, I copy the frame bitmap(raw RGBA byte array) from GPU to CPU, then create tensor by `torch.tensor(rawRgbaByteArray, ...)` (in other words, copy from CPU to GPU). Obviously this is unnecessary.

So, is it possible create `torch.Tensor` from `IDirect3DSurface` directly? That is copy from GPU buffer to another GPU buffer by VRAM copying rather than RAM copying, if I understand correctly.

**Comment by dayo05** [comment]
I don't have enough knowledge of this but I found some reference might related to this:

[Convert Cuda memory to LibTorch tensor](https://stackoverflow.com/questions/77390607/how-to-convert-a-cudaarray-to-a-torch-tensor)
[Create TorchSharp tensor from raw pointer](https://github.com/dotnet/TorchSharp/blob/cb7c7831c9f06980626362830d0a01628ca55782/src/TorchSharp/Tensor/Tensor.cs#L50)

If you mix those with P/Invoke, it seems able to create Tensor without coping entire data to CPU.
